### ISSUE TITLE
Move docs off of Apiary

### ISSUE THREAD

**Issue by timdorr** [timdorr/model-s-api#42]
[Apiary](https://apiary.io/) is a great tool for documenting REST APIs. However, the Tesla API is not true REST and this incongruity results in a lot of confusion. There are two main issues:

- Not all of the API Blueprint format conforms well to the Tesla API. Also, the mock API server is not a proper representation of how the Tesla API operates, leading to even more confusion.
- The API Blueprint, while a great format for REST APIs in general, is not commonly used yet and puts up a barrier to entry for contribution. I would love to get every little bit of the API documented, but I can't do it all myself!

So, I'd like to move the docs off of Apiary and over to Github Pages. We can use the built-in Jekyll or another static site generator. There are quite a few options here. I'm going to get started on the infrastructure ASAP and hopefully we can have this all moved over quickly and easily.

**Comment by mseminatore** [comment]
Happy to help out as I've both used and valued your work!  I am particularly interested in seeing the calendar functionality documented.

What are you proposing as a replacement for the mock server?  I just started work to use a fork of the mock server for testing purposes.

**Comment by timdorr** [comment]
There shouldn't be a mock server. The fact that Apiary has one is simply a side effect of using their service. However, it doesn't properly mock the Tesla API (nor should it; it's non-conforming to REST), so it's misleading to use that one. 

Nor do I think we should be setting up a mock server system. That's what I use VCR for in the ruby parts of this project. That way I can test against the real API surface without continually hammering them with requests.

**Comment by timdorr** [comment]
Started some initial work here: https://timdorr.github.io/model-s-api/

Source lives here: https://github.com/timdorr/model-s-api/tree/master/docs

**Comment by mseminatore** [comment]
What do you record using VCR?  Do you record and playback a full test
pass session against Tesla servers?

**Comment by timdorr** [comment]
Yep, and I only have to record once because VCR immediately saves the request for any subsequent runs. Only if I change how I call the API will trigger a re-recording (which will error out until I clear the cassette file, so that I don't accidentally hammer their servers).

**Comment by mseminatore** [comment]
Gotcha.  I like the basic concept of capture/playback.  But doesn't that
leave your personal info (login credentials, tokens, etc.) in the
cassettes?  I must be missing something.

**Comment by timdorr** [comment]
They're [filtered from the cassette file](https://github.com/timdorr/model-s-api/blob/master/spec/spec_helper.rb#L28-L32)

**Comment by timdorr** [comment]
Welp: https://www.oracle.com/corporate/acquisitions/apiary/index.html

(Swear I didn't know about this ahead of time! Congrats to the Apiary folks!)

**Comment by timdorr** [comment]
After messing around with Jekyll far too much, I'm throwing in the towel and just using Gitbook instead. We use it over on Redux and it's worked very well for us so far. It gives a good reading experience both on their site and here on GitHub (and locally as well). Check it out: https://timdorr.gitbook.io/tesla-json-api/

**Comment by timdorr** [comment]
Update: Put it under my personal domain: https://tesla-api.timdorr.com/ I've also put a notice at the top of the old docs. I'll clear them out and mark this closed as soon as everything is moved over.
